MonkeyWrench.Database/Extensions/DBLogin_Extensions.cs

```diff
@@ -69,33 +71,41 @@ public static DBLogin Login (DB db, string login, string password, string ip4, b
 			return result;
 		}
 
-		public static void LoginOpenId (DB db, LoginResponse response, string email, string ip4)
+		public static void Login (DB db, LoginResponse response, string email, string ip4, List<string> userOrgs, bool useGitHub = false, string gitHubLogin = "")
 		{
-			if (string.IsNullOrEmpty (Configuration.OpenIdProvider) && string.IsNullOrEmpty (Configuration.OauthClientId))
-				throw new Exception ("No OpenId provider available");
+			string [] specs;
 
-			if (string.IsNullOrEmpty (Configuration.OpenIdRoles))
-				throw new Exception ("No OpenId roles specified");
+			string username = useGitHub ? gitHubLogin : email;
```

Comment:
There's a possible attack vector here, if someone manages to create a github account that matches an openid email.

That's currently not possible (`@` isn't allowed in github user names), but there's also no guarantees about the future.

I think the simplest solution for now would be to validate that a github login does not contain a `@`, and at the same time validate that an email does contain a `@`. That way openid and github logins would fall into two disjoint domains.